### PR TITLE
Make share page public to whoever has the link.

### PR DESCRIPTION
**Pull request by karthikv**

### CODE REVIEW

File: routes/auth.js [hunk 1]
```diff
@@ -15,7 +24,8 @@ module.exports = function(app, nconf) {
   app.get('/logout', function(req, res) {
     if (req.session) {
       delete req.session.email;
+      delete req.session.id;
```
Comment (operatorjen) [inline routes/auth.js]: you should be able to delete all session keys with just this one line:

`req.session.reset();`

File: routes/index.js [hunk 2]
```diff
@@ -58,4 +66,54 @@ module.exports = function(app, nconf, db) {
...
+   * extracts the existing scaffolds into the request object This is meant to be
+   * used as Express middleware.
+   * Requires: web request, web response, next function to call when done
+   */
+  function confirmScaffoldExistence(req, res, next) {
+    var numCallbacks = 0;
+    var numFinished = 0;
+
+    if (req.params.projectId) {
+      numCallbacks++;
```
Comment (operatorjen) [inline routes/index.js:80]: nicer formatting to write as:

`numCallbacks ++;` :)

File: routes/index.js [hunk 3]
```diff
@@ -58,4 +66,54 @@ module.exports = function(app, nconf, db) {
...
+          numFinished++;
+          if (numFinished == numCallbacks) {
+            next();
+          }
+        }
+      });
+    }
+
+    if (req.params.screenId) {
+      numCallbacks++;
```
Comment (operatorjen) [inline routes/index.js:100]: same for formatting as above

File: routes/index.js [hunk 4]
```diff
@@ -58,4 +66,54 @@ module.exports = function(app, nconf, db) {
...
+        if (err) {
+          next(err);
+        } else if (!project) {
+          next(new Error('Project could not be found.'));
+        } else {
+          req.project = project;
+
+          // have all callbacks finished?
+          numFinished++;
+          if (numFinished == numCallbacks) {
```
Comment (operatorjen) [inline routes/index.js]: `===`

File: routes/index.js [hunk 5]
```diff
@@ -58,4 +66,54 @@ module.exports = function(app, nconf, db) {
...
+      projects.get(req, db, req.params.projectId, function(err, project) {
+        if (err) {
+          next(err);
+        } else if (!project) {
+          next(new Error('Project could not be found.'));
+        } else {
+          req.project = project;
+
+          // have all callbacks finished?
+          numFinished++;
```
Comment (operatorjen) [inline routes/index.js:91]: same for formatting as above

File: routes/index.js [hunk 6]
```diff
@@ -58,4 +66,54 @@ module.exports = function(app, nconf, db) {
...
+      screens.get(req, db, req.params.screenId, function(err, screen) {
+        if (err) {
+          next(err);
+        } else if (!screen) {
+          next(new Error('Screen could not be found.'));
+        } else {
+          req.screen = screen;
+
+          // have all callbacks finished?
+          numFinished++;
```
Comment (operatorjen) [inline routes/index.js:111]: formatting

File: routes/index.js [hunk 7]
```diff
@@ -58,4 +66,54 @@ module.exports = function(app, nconf, db) {
...
+          if (numFinished == numCallbacks) {
```
Comment (operatorjen) [inline routes/index.js]: `===`

**Comment by operatorjen** [comment]
cool - fix nits and you can merge!

File: routes/index.js (hunk 2 above)
Comment (karthikv) [inline routes/index.js:80]: Any reason why you think this is better? I've actually always seen it written as either `variable++` or `++variable`. I've also been using `variable++` in other commits, so perhaps it's better to keep it consistent.

Comment (operatorjen) [inline routes/index.js:80]: yeah usually it's just for readability .. like for (var i = 0; i < whatever.length; i ++) {

but if it's already like that everywhere else I guess you can leave it.

**Pull request merged**